pkg/server/router/credential.go

```diff
@@ -145,6 +147,140 @@ func (cr CredentialRouter) GetCredential(ctx context.Context, w http.ResponseWri
+// @Description  Get credential status by id
+// @Tags         CredentialAPI
+// @Accept       json
+// @Produce      json
+// @Param        id   path      string  true  "ID"
+// @Success      200  {object}  GetCredentialStatusResponse
+// @Failure      400  {string}  string  "Bad request"
+// @Router       /v1/credentials/{id}/status [get]
+func (cr CredentialRouter) GetCredentialStatus(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
+	id := framework.GetParam(ctx, IDParam)
+	if id == nil {
+		errMsg := "cannot get credential without ID parameter"
+		logrus.Error(errMsg)
+		return framework.NewRequestErrorMsg(errMsg, http.StatusBadRequest)
+	}
+
+	gotCredential, err := cr.service.GetCredential(credential.GetCredentialRequest{ID: *id})
+	if err != nil {
+		errMsg := fmt.Sprintf("could not get credential with id: %s", *id)
+		logrus.WithError(err).Error(errMsg)
+		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusBadRequest)
+	}
+
+	resp := GetCredentialStatusResponse{
+		Revoked: gotCredential.Revoked,
+	}
+
+	return framework.Respond(ctx, w, resp, http.StatusOK)
+}
+
+type GetCredentialStatusListResponse struct {
+	ID            string                        `json:"id"`
+	Credential    *credsdk.VerifiableCredential `json:"credential,omitempty"`
+	CredentialJWT *keyaccess.JWT                `json:"credentialJwt,omitempty"`
+}
+
+// GetCredentialStatusList godoc
+// @Summary      Get Credential Status List
+// @Description  Get credential status list by id
+// @Tags         CredentialAPI
+// @Accept       json
+// @Produce      json
+// @Param        id   path      string  true  "ID"
+// @Success      200  {object}  GetCredentialStatusListResponse
+// @Failure      400  {string}  string  "Bad request"
+// @Router       /v1/credentials/status/{id} [get]
+func (cr CredentialRouter) GetCredentialStatusList(ctx context.Context, w http.ResponseWriter, _ *http.Request) error {
+	id := framework.GetParam(ctx, IDParam)
+	if id == nil {
+		errMsg := "cannot get credential without ID parameter"
+		logrus.Error(errMsg)
+		return framework.NewRequestErrorMsg(errMsg, http.StatusBadRequest)
+	}
+
+	gotCredential, err := cr.service.GetCredentialStatusList(credential.GetCredentialStatusListRequest{ID: *id})
+	if err != nil {
+		errMsg := fmt.Sprintf("could not get credential status list with id: %s", *id)
+		logrus.WithError(err).Error(errMsg)
+		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusBadRequest)
+	}
+
+	resp := GetCredentialStatusListResponse{
+		ID:            gotCredential.ID,
+		Credential:    gotCredential.Credential,
+		CredentialJWT: gotCredential.CredentialJWT,
+	}
+
+	return framework.Respond(ctx, w, resp, http.StatusOK)
+}
+
+type UpdateCredentialStatusRequest struct {
+	ID      string `json:"id" validate:"required"`
+	Revoked bool   `json:"revoked" validate:"required"`
+}
+
+func (c UpdateCredentialStatusRequest) ToServiceRequest() credential.UpdateCredentialStatusRequest {
```

Comment:
I don't think this needs to be exported? 
```suggestion
func (c UpdateCredentialStatusRequest) toServiceRequest() credential.UpdateCredentialStatusRequest {
```